native/packages/react-native-bpk-component-button/src/BpkButton.js

```diff
@@ -0,0 +1,120 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2017 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+ import {
+   View,
+   Text,
+   TouchableHighlight,
+ } from 'react-native';
+ import React from 'react';
+ import PropTypes from 'prop-types';
+ import LinearGradient from 'react-native-linear-gradient';
+
+ import styles from './BpkButton-styles';
+
+ const getTextStyle = ({ type, large, disabled }) => {
+   const textStyle = [styles.base.text];
+   if (styles[type].text) {
```

Comment:
Will throw is someone passes an unknown type.

Reply:
That's fine though? It's enforced via propTypes seems expected that it would trhow

Reply:
I haven't put it in yet, but I was thinking it could use `PropTypes.oneOf` to enforce it

Reply:
PropTypes only throw a warning - it's a bad developer experience if someone adds a button component to their view and it throws because they have a typo e.g. 
```
<BpkButton type="primay" />
```
The above would throw a big red screen with a strange error message about accessing an unknown member and any warnings would be hidden.

Reply:
I think that's great, fail early and fail hard after all. The other option is silently accepting errors and making it more likely that the developer misses the issue

Reply:
Or even better.. throw with a nice error message ;)

i.e. 
```
if (TYPES.indexOf(props.type) === -1) {
  throw new Error(`Unknown button type: ${props.type}`);
}
```

Reply:
Yeah I'm all for that

Reply:
My solution:

``` 
if (!BUTTON_TYPES.includes(type)) {
     throw new Error(`"${type}" is not a valid button type. Valid types are ${BUTTON_TYPES.join(', ')}`);
}
```